### PR TITLE
Rna-Transcription: Handle invalid nucleotides in strand

### LINKED ISSUE #248: rna-transcription has missing test cases
As per x-common's canonical data, some tests which handle invalid input are missing: https://github.com/exercism/x-common/blob/master/exercises/rna-transcription/canonical-data.json#L40-L54

### PR DESCRIPTION
**Pull request by tejasbubane**
As per [x-common canonical data](https://github.com/exercism/x-common/blob/master/exercises/rna-transcription/canonical-data.json#L40-L54)

Fixes #248

### CODE REVIEW

**Comment by matthewmorgan** [comment]
@tejasbubane thanks for doing this.  I am wondering if we shouldn't be throwing an exception here instead of returning `null` though?  What are your thoughts?

File: exercises/rna-transcription/example.js [hunk 1]
```diff
@@ -7,7 +7,14 @@ const DNA_TO_RNA = {
...
-    return dna.replace(/./g, nucleotide => DNA_TO_RNA[nucleotide]);
+    let rna = dna.replace(/./g, nucleotide => DNA_TO_RNA[nucleotide])
+
+    if (rna.length !== dna.length) {
+      // invalid characters in the strand
+      return null;
+    }
+    else {
+      return rna;
+    }
```
Comment (tejasbubane) [inline exercises/rna-transcription/example.js]: Not sure if this is the most elegant solution, but it works 😺

**Comment by tejasbubane** [comment]
@matthewmorgan Yep. I think throwing an exception would be better. Conceptually it is a wrong input and returning `null` makes little sense to the consumer of the API. Also [hamming throws an error](https://github.com/exercism/xecmascript/blob/master/exercises/hamming/example.js#L8) for similar reasons.

**Comment by tejasbubane** [comment]
Strangely, `xruby` changed its behaviour from throwing error to silently returning an empty string. https://github.com/exercism/xruby/pull/358/files. Could not find any history with `x-common` since the file seems to be moved into `exercises` directory.

**Comment by matthewmorgan** [comment]
Are you willing to make the change to throwing an exception then?  If not, I am happy to do it.  Sounds like we agree it should be done.

**Comment by tejasbubane** [comment]
> Were this part of a larger app, that would be a better way to indicate nonsensical input to the consumer.

Agreed. 👍  Will make the change

**Comment by tejasbubane** [comment]
@matthewmorgan fixed.

**Comment by matthewmorgan** [comment]
Lovely!  Thanks again, keep up the good work.

**Pull request merged**